huaweicloud/resource_huaweicloud_rds_read_replica_instance.go

```diff
@@ -0,0 +1,761 @@
+	if err != nil {
+		return err
+	}
+	if instance.Id == "" {
+		d.SetId("")
+		return nil
+	}
+
+	log.Printf("[DEBUG] Retrieved rds read replica instance %s: %#v", instanceID, instance)
+
+	az := readAvailabilityZone(*instance)
+	d.Set("name", instance.Name)
+	d.Set("flavor", instance.FlavorRef)
+	d.Set("region", instance.Region)
+	d.Set("availability_zone", az)
+	d.Set("private_ips", instance.PrivateIps)
+	d.Set("public_ips", instance.PublicIps)
+	d.Set("vpc_id", instance.VpcId)
+	d.Set("subnet_id", instance.SubnetId)
+	d.Set("security_group_id", instance.SecurityGroupId)
+	d.Set("type", instance.Type)
+	d.Set("status", instance.Status)
+	d.Set("enterprise_project_id", instance.EnterpriseProjectId)
+
+	primaryInstanceID, err := readPrimaryInstanceID(*instance)
+	if err != nil {
+		return err
+	}
+	d.Set("primary_instance_id", primaryInstanceID)
+
+	// set tags
+	tagsMap := make(map[string]interface{})
+	for _, tag := range instance.Tags {
+		tagsMap[tag.Key] = tag.Value
+	}
+	d.Set("tags", tagsMap)
+
+	// save volume
+	volumeList := make([]map[string]interface{}, 0, 1)
+	volume := map[string]interface{}{
+		"type":               instance.Volume.Type,
+		"size":               instance.Volume.Size,
+		"disk_encryption_id": instance.DiskEncryptionId,
+	}
+	volumeList = append(volumeList, volume)
+	if err := d.Set("volume", volumeList); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving volume to RDS read replica instance (%s): %s", d.Id(), err)
+	}
+
+	// save database
+	dbList := make([]map[string]interface{}, 0, 1)
+	database := map[string]interface{}{
+		"type":      instance.DataStore.Type,
+		"version":   instance.DataStore.Version,
+		"port":      instance.Port,
+		"user_name": instance.DbUserName,
+	}
+	dbList = append(dbList, database)
+	if err := d.Set("db", dbList); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving data base to RDS read replica instance (%s): %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceRdsReadReplicaInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	config := meta.(*Config)
+	clientV3, err := config.RdsV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating huaweicloud rds v3 client: %s ", err)
+	}
+	clientV1, err := config.RdsV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud rds v1 client: %s ", err)
+	}
+
+	// Fetching node id
+	var nodeID string
+	res := make(map[string]interface{})
+	v, err := fetchRdsReplicaInstanceV3ByList(d, clientV3)
+	if err != nil {
+		return err
+	}
+	if v == nil {
+		return fmt.Errorf("RDS read replica instance: %s was not found", d.Id())
+	}
+
+	res["list"] = v
+	opts := resourceRdsReplicaInstanceV3UserInputParams(d)
+	v, _ = opts["nodes"]
+	v, err = flattenRdsReplicaInstanceV3Nodes(res, nil, v)
+	if err != nil {
+		return err
+	}
+
+	nodes := v.([]interface{})
+	if len(nodes) > 0 {
+		nodeID = nodes[0].(map[string]interface{})["id"].(string)
```

Comment:
should use node id with master role instead of nodes[0]

Reply:
`Nodes` of read replica instance has only one member and who's role is 'readreplica', so we can get it by using `Nodes[0]`.

Reply:
fair enough.